mondrian/src/main/java/mondrian/rolap/SqlMemberSource.java

```diff
@@ -682,7 +682,7 @@ public Object getMemberCacheLock() {
                 hierarchy.addToFrom(sqlQuery, exp);
             }
             final String s = exp.getExpression(sqlQuery);
-            String alias = sqlQuery.addSelect(s, null);
+            String alias = sqlQuery.addSelect(s, property.getSqlType());
```

Comment:
See my comment above. Just pass the type, null or whatever else was defined in the schema.